packages/commonwealth/server/util/validateTopicThreshold.ts

```diff
@@ -0,0 +1,61 @@
+import { WhereOptions } from 'sequelize/types';
+import { ChainNetwork, ChainType } from 'common-common/src/types';
+import TokenBalanceCache from 'token-balance-cache/src/index';
+import { factory, formatFilename } from 'common-common/src/logging';
+
+import { DB } from '../database';
+import { ChainAttributes } from '../models/chain';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+const validateTopicThreshold = async (
+  tbc: TokenBalanceCache,
+  models: DB,
+  topicId: number,
+  userAddress: string
+): Promise<boolean> => {
+  if (!topicId || !userAddress) return true;
+  try {
+    const topic = await models.OffchainTopic.findOne({
+      where: { id: topicId },
+      include: [
+        {
+          model: models.Chain,
+          required: true,
+          as: 'chain',
+          where: {
+            // only support thresholds on token forums
+            // TODO: can we support for token-backed DAOs as well?
```

Comment:
this is still incredibly problematic with no good solution until we get contracts factored out of the `chain` model. looking forward to getting beyond this poor foresight decision from summer 2021 soon.